boost-1.79.0 fails even with latest master.

### Description of the bug

The 2.4.2 as well as current master branch does not build with boost-1.79.0.

Relevant errors:

    /home/piotr/src/PrusaSlicer/src/slic3r/Utils/AppUpdater.cpp:245:52: error: ‘fstream’ is not a member of ‘boost::filesystem’; did you mean ‘std::fstream’?
    /home/piotr/src/PrusaSlicer/src/slic3r/Utils/AppUpdater.cpp:246:33: error: ‘file’ was not declared in this scope
    /home/piotr/src/PrusaSlicer/src/slic3r/GUI/DesktopIntegrationDialog.cpp:189:30: error: variable ‘std::ofstream output’ has initializer but incomplete type
    /home/piotr/src/PrusaSlicer/src/slic3r/GUI/GUI_App.cpp:865:30: error: variable ‘std::ifstream stream’ has initializer but incomplete type


### Project file & How to reproduce

- Update boost to 1.79.0
- Fetch sources
- mkdir build && cd build
- cmake .. -DOPENVDB_FIND_MODULE_PATH="/usr/lib64/cmake/OpenVDB" -DSLIC3R_FHS=ON -DSLIC3R_GTK=3 -DSLIC3R_GUI=ON -DSLIC3R_PCH=OFF -DSLIC3R_STATIC=OFF -DSLIC3R_WX_STABLE=ON -Wno-dev
- make -j$(nproc)
- see errors.

### Checklist of files included above

- [ ] Project file
- [ ] Screenshot

### Version of PrusaSlicer

2.4.2, master

### Operating system

Linux, Gentoo, ~amd64

### Printer model

mk3s+

--- a/src/slic3r/GUI/DesktopIntegrationDialog.cpp
+++ b/src/slic3r/GUI/DesktopIntegrationDialog.cpp
@@ -10,6 +10,7 @@
 #include "libslic3r/Platform.hpp"
 #include "libslic3r/Config.hpp"
 
+#include <boost/nowide/fstream.hpp>
 #include <boost/filesystem.hpp>
 #include <boost/log/trivial.hpp>
 #include <boost/dll/runtime_symbol_info.hpp>

--- a/src/slic3r/GUI/GUI_App.cpp
+++ b/src/slic3r/GUI/GUI_App.cpp
@@ -13,6 +13,7 @@
 #include <cstdlib>
 #include <regex>
 #include <string_view>
+#include <boost/nowide/fstream.hpp>
 #include <boost/algorithm/string/predicate.hpp>
 #include <boost/algorithm/string.hpp>
 #include <boost/format.hpp>

--- a/src/slic3r/Utils/AppUpdater.cpp
+++ b/src/slic3r/Utils/AppUpdater.cpp
@@ -5,6 +5,7 @@
 
 #include <boost/filesystem.hpp>
 #include <boost/log/trivial.hpp>
+#include <boost/nowide/fstream.hpp>
 #include <boost/nowide/convert.hpp>
 #include <boost/property_tree/ini_parser.hpp> 
 #include <curl/curl.h>
@@ -242,7 +243,7 @@ boost::filesystem::path AppUpdater::priv::download_file(const DownloadAppData& d
 			tmp_path += format(".%1%%2%", get_current_pid(), ".download");
 			try
 			{
-				boost::filesystem::fstream file(tmp_path, std::ios::out | std::ios::binary | std::ios::trunc);
+				boost::nowide::fstream file(tmp_path, std::ios::out | std::ios::binary | std::ios::trunc);
 				file.write(body.c_str(), body.size());
 				file.close();
 				boost::filesystem::rename(tmp_path, dest_path);

